analyze_map_test.go

```diff
@@ -82,6 +82,16 @@ func TestAnalyzeMapBody(t *testing.T) {
 			},
 			term: ";",
 		},
+		"valid types": {
```

Comment:
Do we need a negative test to ensure this type doesn't allow "special parameters"?

Reply:
added negative test case